### ISSUE TITLE
Tilix doesnt open in the folder from where it is split

### ISSUE THREAD

**Issue by deepu105** [gnunn1/tilix#941]
When Tilix is opened I get a prompt saying there is an issue with configuration and the prompt links to https://gnunn1.github.io/tilix-web/manual/vteconfig/
But I have already done these configuration and everything was working fine. I guess this started happening after some of the libraries in my system was updated.
And now Tilix doesnt open from the same folder where it was split.
I tried all three options mentioned but still no luck. 
I guess the issue come from somewhere else as I checked if `__vte_osc7` is being called and it was called. I even added an echo in that method and whenever i split it prints the home folder from `echo "\033]7;file://%s%s\007" "${HOSTNAME:-}" "$(__vte_urlencode "${PWD}")"` instead of current dir

I know this might not be the correct forum for this but I spend a lot of time trying to figure it out I thought its worth a shot asking here. sorry for the trouble 

I'm on Fedora 4.10.16-200.fc25.x86_64
Zsh version 5.2
VTE version 4601
TERM xterm-256color
Tilix version 1.5.8-1.fc25

**Comment by gnunn1** [comment]
Did this start updating with the upgrade of tilix or did it happen with the previous version as well? 

If you switch to bash as a troubleshooting issue does it start working as expected?

**Comment by gnunn1** [comment]
Oh, and in gnome-terminal, if you open a new tab does it open in the same directory as the previous tab? If no that would indicate it's a more general issue you are facing rather then a tilix specific issue.

**Comment by deepu105** [comment]
The problem happens in gnome-terminal as well so I guess its something with VTE or ZSH in general. Any idea where I should start?

**Comment by b3rew** [comment]
I don't know why this issue is closed maybe it's not related to tilix
it's gnome-terminal  bug https://bugs.launchpad.net/ubuntu-gnome/+bug/1193993

but you can simply fix it by adding this to `.bashrc` or `.zshrc`

```sh
. /etc/profile.d/vte.sh
```

**Comment by deepu105** [comment]
I dont think its a tilix issue. I think Im having some other gnome bug. I
always get home for new terminal after a restart but if I put machine on
sleep and resume it works fine.

**Comment by phw** [comment]
@b3rew The issue you describe is well documented in the Wiki: https://github.com/gnunn1/tilix/wiki/VTE-Configuration-Issue . Also Tilix detects this issue and directs the user to that Wiki page.

**Comment by albfan** [comment]
This looks to be still a thing in 2021:

Here output of `PROMPT_COMMAND` showing special chars with `cat -A -`. This is working on gnome-terminal and not working on tilix:

```
[alberto@localhost ~]$ echo $($PROMPT_COMMAND)| cat -A -
^[]777;notify;Command completed;echo $($PROMPT_COMMAND)| cat -A -^[\^[]777;precmd^[\^[]0;alberto@fedora:~^[\^[]7;file://fedora/home/alberto^[\$
[alberto@localhost ~]$ cd Descargas/
[alberto@localhost Descargas]$ echo $($PROMPT_COMMAND)| cat -A -
^[]777;notify;Command completed;echo $($PROMPT_COMMAND)| cat -A -^[\^[]777;precmd^[\^[]0;alberto@fedora:~/Descargas^[\^[]7;file://fedora/home/alberto/Descargas^[\$
```

Any ideas?

**Comment by albfan** [comment]
check #1954

TL;DR:

HOSTNAME=""
looks like HOSTNAME changed from:

$ /usr/bin/hostname 2>/dev/null
localhost
to:

$ /usr/bin/hostnamectl --transient 2>/dev/null
fedora
and that puzzles tilix

**Comment by albfan** [comment]
Probably problem is systemd network: https://www.freedesktop.org/software/systemd/man/systemd.network.html

> UseHostname=
> When true (the default), the hostname received from the DHCP server will be set as the transient hostname of the system.

that's why none of this work to change transient: https://www.freedesktop.org/software/systemd/man/systemd.network.html

Here details:

https://www.linode.com/community/questions/9524/solved-cannot-set-transient-hostname-on-arch-linux-linode

Honestly I prefer to change HOSTNAME than touch network config. See gnome terminal works with any hostname value, so probably tilix should use same way to get hostname as gnome-terminal

**Comment by albfan** [comment]
Final solution thanks to @ckulacoglu

https://sleeplessbeastie.eu/2020/06/22/how-to-change-system-hostname/
Transient hostname can be used only in a specific situation where persistent hostname is set to localhost, localhost.localdomain or ends with these names (like debian.localhost.localdomain). See additional notes at the end.

So as long your hostname is set to localhost, your transient hostname might be different from your hostname (set by dhcp etc), causing the issues. Giving your system a personal hostname does not affect how "localhost" resolution works (which points to 127.0.0.1) unless you manually mess up with /etc/hosts

hostnamectl set-hostname thinkpad

fix this for me without doing anything on ~/.bashrc

**Comment by AdnanHodzic** [comment]
> I don't know why this issue is closed maybe it's not related to tilix it's gnome-terminal bug https://bugs.launchpad.net/ubuntu-gnome/+bug/1193993
> 
> but you can simply fix it by adding this to `.bashrc` or `.zshrc`
> 
> ```shell
> . /etc/profile.d/vte.sh
> ```

In 2021 with Ubuntu 21.10 solution should be:

Add following line to your `.bashrc`:

```
. /etc/profile.d/vte-2.91.sh 
```

**Comment by albfan** [comment]
@AdnanHodzic reporter close the issue (not anyone in tilix crew) That script is part of the solution but then hostname transient still can cause problems. You need to confirm your hostname is not transient (see previous comments)

**Comment by AdnanHodzic** [comment]
> @AdnanHodzic reporter close the issue (not anyone in tilix crew) That script is part of the solution but then hostname transient still can cause problems. You need to confirm your hostname is not transient (see previous comments)

@albfan no, I don't have a transient hostname, and I even ran `hostnamectl set-hostname myhostname` but nothing changed. Only when I do as I specified:

> In 2021 with Ubuntu 21.10 solution should be:
> 
> Add following line to your `.bashrc`:
> 
> ```
> . /etc/profile.d/vte-2.91.sh 
> ```

Tilix will split terminals from their current location.

**Comment by albfan** [comment]
@AdnanHodzic this is causing noise, you might or mightnot have problems with hostname, but source that script is mandatory. (that's out of question)

**Comment by juxuny** [comment]
> > I don't know why this issue is closed maybe it's not related to tilix it's gnome-terminal bug https://bugs.launchpad.net/ubuntu-gnome/+bug/1193993
> > but you can simply fix it by adding this to `.bashrc` or `.zshrc`
> > ```shell
> > . /etc/profile.d/vte.sh
> > ```
> 
> In 2021 with Ubuntu 21.10 solution should be:
> 
> Add following line to your `.bashrc`:
> 
> ```
> . /etc/profile.d/vte-2.91.sh 
> ```

This work for my on Ubuntu 22.04
